lib/logstash/codecs/avro.rb

```diff
@@ -131,4 +184,170 @@ def encode(event)
       @on_event.call(event, buffer.string)
     end  
   end
+
+  private
+  def fetch_schema(uri_string)
+    http_connection = uri_string.start_with?('http://')
+    https_connection = uri_string.start_with?('https://')
+
+    if http_connection
+      ssl_config_provided = original_params.keys.select {|k| k.start_with?("ssl_") && k != "ssl_enabled" }
+      if ssl_config_provided.any?
+        raise_config_error! "When SSL is disabled, the following provided parameters are not allowed: #{ssl_config_provided}"
+      end
+
+      credentials_configured = @username && @password
+      @logger.warn("Credentials are being sent over unencrypted HTTP. This may bring security risk.") if credentials_configured
+      fetch_remote_schema(uri_string)
+    elsif https_connection
+      validate_ssl_settings!
+      fetch_remote_schema(uri_string)
+    else
+      # local schema
+      URI.open(uri_string, &:read)
+    end
+  end
+
+  def fetch_remote_schema(uri_string)
+    client = nil
+    client_options = {}
+
+    if @proxy && !@proxy.empty?
+      client_options[:proxy] = @proxy.to_s
+    end
+
+    basic_auth_options = build_basic_auth
+    client_options[:auth] = basic_auth_options unless basic_auth_options.empty?
+
+    if @ssl_enabled
+      ssl_options = build_ssl_options
+      client_options[:ssl] = ssl_options unless ssl_options.empty?
+    end
+
+    client = Manticore::Client.new(client_options)
+
+    body = Stud.try(3.times, [Manticore::SocketException, Manticore::Timeout, StandardError]) do
```

Comment:
how about catching Manticore::ManticoreException for broader network related exceptions and being more specific for BadResponseCodeError? It requires to create BadResponseCodeError